### ISSUE TITLE
Can ScaLR be used for encoding xyz pointclouds?

### ISSUE THREAD

**Issue by Nik-V9** [valeoai/ScaLR#2]
Hi, Thanks for sharing this great work!

I was wondering if ScaLR would work on pointclouds which don't have laser return intensity and are not obtained from a LiDAR but instead lets say depth maps or some other SfM method?

If yes, what is the recommended dummy value to feed in for the laser return intensity?

Looking forward to your thoughts!

**Comment by gpuy** [comment]
Hi, thanks for your positive comment!

Concerning your question, if you mean pretraining a new backbone on your own data using the method ScaLR, then yes it should work. There may be better choices of backbones though, e.g., if you have indoor data.

If, instead, you mean re-using the checkpoint we provide on your data, it will be highly dependent on the type of data. 

For eg indoor data, the provided checkpoint won't work: WaffleIron hyperparameters are tuned for driving scenes and pretraining was only done on driving scenes.

If your depth data are extracted from, eg, images from driving datasets, then there is more hope, provided that you have metric depth  (or are close enough to the true scale). I haven't tried yet. It would be interesting to do so.

Hard to tell more without more context but I hope it answers your question.

**Comment by gpuy** [comment]
The best in any case is if you can afford to retrain using this new type of data

**Comment by Nik-V9** [comment]
I see, Thanks for the response!

I was asking in the context of using it with metric depth maps (probably outdoor). Do you think 128 might be a good dummy value to pad for experimentation?

**Comment by gpuy** [comment]
In the code, the intensities are normalized ([here](https://github.com/valeoai/ScaLR/blob/main/datasets/pc_dataset.py#L114)) to have (approximately) mean 0 and standard deviation 1 on each dataset.

To apply the model on a new dataset without intensity, I would, e.g., copy the template of [`NuScenesSemSeg`](https://github.com/valeoai/ScaLR/blob/main/datasets/nuscenes_for_scalr.py#L42) where I would
1. set [`self.mean_int`](https://github.com/valeoai/ScaLR/blob/main/datasets/nuscenes_for_scalr.py#L67) to 0
2. set [`self.std_int`](https://github.com/valeoai/ScaLR/blob/main/datasets/nuscenes_for_scalr.py#L68) to 1
3. load a point cloud [`pc`](https://github.com/valeoai/ScaLR/blob/main/datasets/nuscenes_for_scalr.py#L107) as an array of size `N x 4` where the first three columns are x, y, z coordinates and the last column (intensity) is filled with zeros. After normalisation the intensity will stay at zero (the average intensity that the network should see). The z-axis should be the vertical axis.

**Comment by Nik-V9** [comment]
Thanks for the detailed steps, will try this out!
